notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/InstallationTemplate.java

```diff
@@ -97,6 +102,69 @@ public void removeHeader(String name) {
         mHeaders.remove(name);
     }
 
+    /**
+     * Serialize InstallationTemplate to JSONObject.
+     *
+     * @param name Name of template
+     * @param installationTemplate The templates that should no longer be in the collection.
+     * @return serialized templateObject.
+     */
+    public static JSONObject serialize(String name, InstallationTemplate installationTemplate) {
+        JSONObject templateObject = new JSONObject();
+        try {
+            templateObject.put("name", name);
+            templateObject.put("body", installationTemplate.getBody());
+            JSONObject headers = new JSONObject();
+            Iterable<Map.Entry<String, String>> headersIterators = installationTemplate.getHeaders();
+            for(Map.Entry<String, String> header: headersIterators){
+                headers.put(header.getKey(), header.getValue());
+            }
+            templateObject.put("headers", headers);
+            JSONArray tagsArray = new JSONArray();
+            for (String tag : installationTemplate.getTags()) {
+                tagsArray.put(tag);
+            }
+            templateObject.put("tags", tagsArray);
+        } catch (JSONException ex) {
+            // Investigating the possible sources of JSONException, it seems the only time it is
+            // thrown is when a Number is invalid. Because we are exclusively serializing strings
+            // above, this exception _should_ never be thrown.
+            throw new RuntimeException("Invalid template, unable to serialize", ex);
+        }
+        return templateObject;
+    }
+
+    /**
+     * Deserialize JSONObject to InstallationTemplate.
+     *
+     * @param installationTemplate Template to deserialize
+     * @return Deserialized template.
+     * @throws JSONException When there's a schema-mismatch of the object to populate, and what
+     * appears in the serialized form of the template.
+     */
+    public static InstallationTemplate deserialize(JSONObject installationTemplate) throws JSONException {
```

Comment:
Same here, let's make it package-private.

Reply:
fixed